src/compile.ts

```diff
@@ -35,7 +36,7 @@ export async function compile(
   const solcInput = {
     language: "Solidity",
     sources: sources,
-    settings: compilerSettings,
+    settings: {...compilerSettings, ...solcSettings},
```

Comment:
People shouldn't be allowed to change the `outputSelection` setting. I would suggest changing the global `compilerSettings` variable to be an `outputSelection` variable with only the `outputSelection` part of the settings, so that this line can be changed to:

```suggestion
    settings: { ...solcSettings, outputSelection },
```

Reply:
@frangio: Well, you'd also need to change this:
```
const compilerSettings = {
  outputSelection: {
    '*': {
      '': [
        'ast',
      ],
    },
  },
};
```
To this:
```
const outputSelection = {
  '*': {
    '': [
      'ast',
    ],
  },
};
```
And the bottom-line result would be equivalent to my original suggestion... wouldn't it? Or what exactly am I missing here?

Reply:
Yes the variable needs to be changed.

The result would be different in the case where the user includes a value for `outputSelection` in the `solc-settings` flag. In the expression `{...compilerSettings, ...solcSettings}` JavaScript will keep the last value of `outputSelection` that it sees. Since `solcSettings` is last, if it contains `outputSelection` its value will be in the final object. By switching the order we ensure that we always keep our custom value for the output selection.

Reply:
@frangio: Right, I've missed that, thanks for explaining :)
I'll fix the other part too (declare `outputSelection` instead of `compilerSettings`).